nearswap/src/storage_management.rs

```diff
@@ -0,0 +1,110 @@
+use near_contract_standards::storage_management::{
+    StorageBalance, StorageBalanceBounds, StorageManagement,
+};
+use near_sdk::{assert_one_yocto, env, near_bindgen, AccountId, Balance};
+use std::convert::TryInto;
+use std::collections::HashMap;
+
+use crate::*;
+use crate::constants::*;
+
+/// Implements users storage management for NearSwap.
+#[near_bindgen]
+impl StorageManagement for NearSwap {
+
+    // Register the caller and store minimal deposit.
+    #[payable]
+    fn storage_deposit(
+        &mut self,
+        account_id: Option<ValidAccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance {
+        let amount = env::attached_deposit();
+        let account_id = account_id
+            .map(|a| a.into())
+            .unwrap_or_else(|| env::predecessor_account_id());
```

Comment:
this is wrong. If the original account_id validation fails, we should panic, rather than falling back to `predecessor_account_id`.
Here add the following test case:
```
storage_deposit(Some("123"), None)  // this should fail
```

Reply:
continued on discord